### PR TITLE
Refactor attachments resource

### PR DESCRIPTION
**Pull request by jonathanstowell**

### CODE REVIEW

File: src/ZendeskApi.Client/Extensions/HttpResponseExtensions.cs [hunk 1]
```diff
@@ -1,47 +1,129 @@
...
+using Newtonsoft.Json;
 using ZendeskApi.Client.Exceptions;
 
 namespace ZendeskApi.Client.Extensions
 {
-    public static class HttpResponseExtensions
+    internal static class HttpResponseExtensions
     {
-        public static async Task ThrowIfUnsuccessful(
+        internal static async Task<HttpResponseMessage> IsNullWhen(
```
Comment (LoadData) [inline src/ZendeskApi.Client/Extensions/HttpResponseExtensions.cs]: Should this not be returning a bool, with the name starting with `Is`?

Comment (jonathanstowell) [reply src/ZendeskApi.Client/Extensions/HttpResponseExtensions.cs]: Nope. If you look at the usage I want to chain together a series of actions after a request so I can compose together flows for slightly different usages: https://github.com/justeat/ZendeskApiClient/pull/195/files#diff-c0163894d98e12399628d6a8ef46031cR69

Comment (jonathanstowell) [reply src/ZendeskApi.Client/Extensions/HttpResponseExtensions.cs]: As it then lets me reuse the functions for requests which don't quite fit the normal pattern: https://github.com/justeat/ZendeskApiClient/pull/195/files#diff-e167f1e3a33337c1008dc6d0b9419135R37

Comment (corneliuskopp) [reply src/ZendeskApi.Client/Extensions/HttpResponseExtensions.cs]: Might still be preferable to call it `TransformNullIf()` or `MakeNullIf()`; i'd concur that `IsWhatever` generally implies a bool-typed member.

Comment (jonathanstowell) [reply src/ZendeskApi.Client/Extensions/HttpResponseExtensions.cs]: Changed it to SetToNullWhen which represents what its doing more accurately?

**Pull request merged**